src/components/core/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,280 @@
+import {
+  FiCalendar as CalendarIcon,
+  FiChevronLeft as ChevronLeftIcon,
+  FiChevronRight as ChevronRightIcon,
+  FiX,
+} from "react-icons/fi";
+
+import Button from "components/core/Button/Button";
+import Icon from "components/core/Icon/Icon";
+import Input from "components/core/Input/Input";
+import Text from "components/core/Text/Text";
+import {
+  PrimitiveDatePicker,
+  PrimitiveDatePickerClearTrigger,
+  PrimitiveDatePickerContent,
+  PrimitiveDatePickerControl,
+  PrimitiveDatePickerInput,
+  PrimitiveDatePickerLabel,
+  PrimitiveDatePickerNextTrigger,
+  PrimitiveDatePickerPositioner,
+  PrimitiveDatePickerPrevTrigger,
+  PrimitiveDatePickerRangeText,
+  PrimitiveDatePickerTable,
+  PrimitiveDatePickerTableBody,
+  PrimitiveDatePickerTableCell,
+  PrimitiveDatePickerTableCellTrigger,
+  PrimitiveDatePickerTableHead,
+  PrimitiveDatePickerTableHeader,
+  PrimitiveDatePickerTableRow,
+  PrimitiveDatePickerTrigger,
+  PrimitiveDatePickerView,
+  PrimitiveDatePickerViewControl,
+  PrimitiveDatePickerViewTrigger,
+} from "components/primitives";
+
+import type { PrimitiveDatePickerProps } from "components/primitives/DatePicker/DatePicker";
+import type { ReactNode } from "react";
+
+export interface Props extends PrimitiveDatePickerProps {
+  /** Input label. */
+  label?: ReactNode;
+  /** Render to open date picker. */
+  trigger?: ReactNode;
+  /** Date input field. */
+  input?: ReactNode;
+  /** Render to clear date input. */
+  clearTrigger?: ReactNode;
+}
+
+/**
+ * Date picker.
+ */
+const DatePicker = ({
+  label,
+  trigger,
+  input,
+  clearTrigger,
+  ...rest
+}: Props) => (
+  <PrimitiveDatePicker positioning={{ sameWidth: true }} {...rest}>
+    {label && <PrimitiveDatePickerLabel>{label}</PrimitiveDatePickerLabel>}
+
+    <PrimitiveDatePickerControl>
+      {input || (
+        <PrimitiveDatePickerInput asChild>
+          <Input />
+        </PrimitiveDatePickerInput>
+      )}
+
+      <PrimitiveDatePickerTrigger asChild aria-label="Open date picker">
+        {trigger || (
+          <Button variant="secondary" borderColor="fg.default">
```

Comment:
Not sure what the best approach for this is (we have to do this a lot downstream in MS as well) but we should add `h={10}` so that the default trigger `Button` matches the height of the default input (and maybe the clear trigger too?). Down to normalize sizing across components though too, and in that case probs best for a separate PR. Let me know your thoughts!